### ISSUE TITLE
gdbus.exe has stopped working - error message after closing a Cherrytree file in version 1.2.0

### ISSUE THREAD

**Issue by Atlntssplayer** [giuspen/cherrytree#2579]
**Version, Operating system**
Cherrytree version 1.2.0 (both nolatex and normal portable releases)
Windows 7 x64

**Describe the bug**
Using Cherrytree version 1.2.0 after closing a .ctd file I get the error message from Windows that "gdbus.exe has stopped working".
This happens even after just opening and closing a Cherrytree document without modifying anything in it.
Happens with the nolatex and normal versions too.
I only tested the portable releases, cherrytree_1.2.0.0_win64_portable.7z and cherrytree_1.2.0.0_win64_portable_nolatex.7z.
I also tested the following older versions for this bug and it didn't happen: 1.1.4, 1.1.3, 1.1.2, 0.39.4
I tested with 2 different documents, a smaller one with about 20 nodes and a big one with about 1900 nodes and the bug happened with both.

**To Reproduce**
If applicable, attach a non-personal document where the issue can be reproduced systematically.
1. Open .ctd file with Cherrytree version 1.2.0
2. Close Cherrytree
3. gdbus.exe has stopped working error happens.

**Comment by giuspen** [comment]
Hello @Atlntssplayer and thanks for reporting.
I'm not seeing this on Windows 11, I will try later on a Windows 10.
gdbus.exe / DBus ( https://en.wikipedia.org/wiki/D-Bus ) is used from cherrytree to communicate between the first and the subsequent instances. Typically every instance (when you open a file or just start from the menu) tries first to talk to an already existing instance to "pass the ball" and exit instead of having many independent instances running.
It's not something you should worry too much because the worse that can happen is that you have more than one instance of cherrytree running on the same file independently, but if I can reproduce it I will look at why is doing so.
Unfortunately I don't have a Windows 7, I'll see if we can reproduce at least on a Windows 10.

**Comment by giuspen** [comment]
Also tested on a Windows 10, could not replicate the issue.

**Comment by Atlntssplayer** [comment]
Thanks for looking into it. If it can't corrupt the opened file than I'm not worried.

I have WinDbg installed and used it before to get a stack trace for devs of another software that was crashing.
Maybe I can use it in this case too if you can't reproduce the problem on your end.
I would need symbol files for it though. I'm not a dev myself so I don't really know how things work.
Anyway if I can help more feel free to tell me what to do.
I love Cherrytree. :)

**Comment by giuspen** [comment]
No worries @Atlntssplayer as it's not dangerous and probably only happening on Windows 7, if you are curious yourself and want to dig this further, you would have to try and build/run from the sources as described in https://github.com/giuspen/cherrytree/blob/master/BUILDING.md#building-cherrytree-on-windows - Take care ;)

**Comment by metal450** [comment]
I'm seeing this too, on Windows 10 1809. gdbus.exe crashes shortly after exiting CherryTree 1.2.0 each time.

**Comment by metal450** [comment]
Also: I'm using a multi-file db, not ctd

**Comment by giuspen** [comment]
Thanks for reporting @metal450 are you also using a portable version or you have it installed?

**Comment by metal450** [comment]
Installed

**Comment by csttsn** [comment]
Having the same issue, W11 Pro with all latest updates, portable version, gdbus.exe crashes after closing the Cherrytree (you can view the error description in "Problem Reports" in the Control Panel).

**Comment by csttsn** [comment]
> I copied the following files from `1.1.4` to `1.2.0`:

Tried replacing only `libglib-2.0-0.dll` with the version from 1.4, but the program does not start. So I decided to revert back to 1.1.4 for now, which is working flawlessly. I hope [giuspen](https://github.com/giuspen) will manage to find the cause of this crash.

**Comment by giuspen** [comment]
@B-Down if you take only gdbus.exe from 1.1.4 and overwrite it into 1.2.0, is the crash still happening?
Your help here is very useful since I cannot see the crash myself, it seems to me clear it's a library/msys2 platform component issue and if you help me to cherry-pick the minimum older component to revert I can then submit here a new bundle so that who has the crash can try it and confirm that it works for them too.

**Comment by csttsn** [comment]
> Well, you quoted one line from my message and seemed to have ignored the rest of it (I listed all the files I replaced **and** specified that I _didn't_ touch `libglib`), so no wonder you didn't get the same result...

I just wanted to test whether replacing this one file will solve the issue, since it is mentioned in the crash log, but looks like it is a part of the larger component that was updated, since, for example, additionally replacing `gdbus.exe` does not help either. I'm amazed you managed to quickly figure out the full list of files to be replaced to avoid the crash.

**Comment by giuspen** [comment]
Anybody experiencing this issue after closing cherrytree, could try https://www.giuspen.net/software/cherrytree_1.2.0.0+13_win64_portable.7z and report if it still happens?

**Comment by metal450** [comment]
Looks like that fixes it

**Comment by giuspen** [comment]
I didn't really apply any change for this issue in cherrytree, I simply noticed that msys2 released a new glib2 package https://github.com/msys2/MINGW-packages/commits/master/mingw-w64-glib2 (together with other updates) and rebuilt cherrytree with that. Unfortunately it doesn't look like windows 7 is supported anymore https://www.msys2.org/docs/windows_support/

**Comment by giuspen** [comment]
Actually I'm wrong, it is stated:
`Mingw Toolchains: MINGW32/MINGW64 environments allow targeting Windows 7+ still. All others allow targeting Windows 8.1+.`
And CherryTree is MINGW64 it should still be supported.

**Comment by metal450** [comment]
Should we be unzipping the build above over our installed CherryTree for now? Is that the best approach (until there's a new version with the fix)?

**Comment by giuspen** [comment]
@metal450 yes you can use https://www.giuspen.net/software/cherrytree_1.2.0.0+13_win64_portable.7z for the time being if you don't want to be nagged from the error dialog.
@B-Down I will prepare a new development build with the latest msys2 distribution and link it here, let me know if it works any better. If not you should raise an issue with them reporting that it doesn't work anymore with Win7 while the website claims it should with MINGW64.
I don't have a Win7 to test so I'm not the right person to raise such issue.

**Comment by metal450** [comment]
Ok great thanks, will do.

Yeah the constant crash dialogs are a bit impactful, since it pops up after a short delay often when you're already typing in another program - & hitting "space" will then launch the full Visual Studio debugger.

**Comment by giuspen** [comment]
Latest build with up to date msys2 packages: https://www.giuspen.net/software/cherrytree_1.2.0.0+20_win64_portable.7z
Please report if the gdbus.exe crash is confirmed gone (as it seemed already in 1.2.0.0+13 above) and if Win7 build is still broken.

**Comment by metal450** [comment]
Confirmed the crash is gone (on Win10)

**Comment by csttsn** [comment]
Hello @giuspen,
Just FYI, I installed the newly released version 1.3.0, and the crash is gone. Thank you for fixing this issue, love CherryTree so much! I wish you all the best!

**Comment by giuspen** [comment]
Hi @csttsn thanks for the feedback, take care 😊
